src/data/FilterCondition.py

```diff
@@ -51,6 +59,8 @@ def __repr__(self) -> str:
         return f"FilterCondition({self.filter_string})"
 
     def get_validated_token(self, token):
+        if is_tags_token(token):
+            return "v['tags']" + token[4:]
```

Comment:
So what's causing the test to fail is that v['tags']['state'] is being evaluating even when tags has no 'state' key. Changing it to "v['tags'].get(" + token[5:-1] +", None)" would probably work. If you can come up with a cleaner way to handle missing keys, use that instead.